Source/BuildControl.cs

```diff
@@ -718,16 +718,22 @@ public void LoadCraft (string filename, string flagname)
 				Debug.LogWarning ($"File '{filename}' does not exist");
 				return;
 			}
-			string craftText = File.ReadAllText (filename);
-			ConfigNode craft = ConfigNode.Parse (craftText);
-			ReplaceLaunchClamps (craft);
 
-			state = State.Planning;
-			craftName = Localizer.Format (craft.GetValue ("ship"));
-			if ((buildCost = getBuildCost (craft, craftText)) != null) {
-				craftConfig = craft;
+			try {
+				string craftText = File.ReadAllText(filename);
+				ConfigNode craft = ConfigNode.Parse(craftText);
+				ReplaceLaunchClamps(craft);
+
+				state = State.Planning;
+				craftName = Localizer.Format(craft.GetValue("ship"));
+				if ((buildCost = getBuildCost(craft, craftText)) != null) {
+					craftConfig = craft;
+				}
+				PlaceCraftHull();
+			} catch (Exception e) {
+				Debug.LogException(e);
+				state = State.Idle;
 			}
-			PlaceCraftHull ();
```

Comment:
I would rather not have a try/catch block at all. First, any mods causing exceptions really need to be fixed. Second, the try block covers too much code. Finally, the real problem is that the setting of `state` and `craftName` should come after the call to `buildCost` (possibly in the conditional block, undecided) rather than before: that way, if there is an exception, they won't get set and won't need to be reset.